src/components/favorites/favorites.js

```diff
@@ -0,0 +1,122 @@
+        } else {
+            localStorage.setItem(keyColor, "1");
+            console.log(`value: ${keyColor} set`);
+        }
+    } else {
+        console.log(`localStorage disabled`);
+    }
+    scoreboard();
+};
+
+function scoreboard() {
+    let scoreboard = "** SCOREBOARD **\n";
+
+    if (storageAvailable("localStorage")) {
+        if (Object.keys(localStorage).length === 0) {
+            scoreboard = `${scoreboard} no scores available`;
+        } else {
+            Object.getOwnPropertyNames(localStorage).forEach( keyColor => {
+                scoreboard = scoreboard + "\n" + `${keyColor}: ${localStorage[keyColor]}`;
+            });
+        };
+    } else {
+        console.log(`localStorage disabled`);
+    }
+    document.querySelector("#scoreboard").value = scoreboard;
+};
+
+function cleanStorage() {
+    localStorage.clear();
+    document.querySelector("#scoreboard").value = "** SCOREBOARD **\n";
+    console.log("Storage clear");
+};
+
+
+export function createFavorites () {
+
+    const container = createElement("div", {
+        className: "container",
+        children:
+        [
+            createElement("div",{
+                className: "hearts",
+                children:
+                [
+                    createElement("img", {
+                        src: heartOrange,
+                        alt: "orange-hearth",
+                        onclick: (event) => {
+                            heartCounter(event);
+                        },
+                    }),
+                    createElement("img", {
+                        src: heartBlue,
+                        alt: "blue-hearth",
+                        onclick: (event) => {
+                            heartCounter(event);
+                        },
+                    }),
+                    createElement("img", {
+                        src: heartPink,
+                        alt: "pink-hearth",
+                        onclick: (event) => {
+                            heartCounter(event);
+                        },
+                    }),
+                    createElement("img", {
+                        src: heartYellow,
+                        alt: "yellow-hearth",
+                        onclick: (event) => {
+                            heartCounter(event);
+                        },
+                    }),
+                    createElement("img", {
+                        src: heartGreen,
+                        alt: "green-hearth",
+                        onclick: (event) => {
+                            heartCounter(event);
+                        },
+                    }),
+                    createElement("img", {
+                        id: "trash",
+                        src: trashCan,
+                        alt: "Delete everything!",
+                        onclick: () => {
+                            cleanStorage();
+                        },
+
+                    }),
+                ],
```

Comment:
You could refator your code to something like that, because you have written a lot of the same code
pls review and check (especially the closing tags ;-))

```suggestion
export function createFavorites() {
  const imgChildren = [
    { src: "heartOrange", alt: "heart-Orange" },
    { src: "heartBlue", alt: "heart-Blue" },
    { src: "heartPink", alt: "heartPink" },
    { src: "heartYellow", alt: "heart-Yellow" },
    { src: "heartGreen", alt: "heart-Green" },
    {
      id: "trashCan",
      src: "trashCan",
      alt: "Delete everything!",
      onclick: () => {
        cleanStorage();
      },
    },
  ];
  const container = createElement("div", {
    className: "container",
    children: [
      createElement("div", {
        className: "hearts",
        children: imgChildren.map(
          ({
            id = false,
            src,
            alt,
            onclick = (event) => {
              heartCounter(event);
            },
          }) => {
            return id
              ? createElement("img", {
                  id,
                  src,
                  alt,
                  onclick: () => {
                    cleanStorage();
                  },
                })
              : createElement("img", {
                  src,
                  alt,
                  onclick: (event) => {
                    heartCounter(event);
                  },
                });
          }
        ),
      }),
      createElement("textarea", {
        className: "output",
        id: "scoreboard",
      }),
    ],
  });
}
```

Reply:
Good idea! 
But I would try to avoid this special case handling with the trash can.
In addition, you could profit by using [Spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) and directly use the function in your `onclick`.

```suggestion
export function createFavorites () {
    const hearts = [
      { src: heartOrange, alt: "Orange Heart" },
      { src: heartBlue, alt: "Blue Heart" },
      { src: heartPink, alt: "Pink Heart" },
      { src: heartYellow, alt: "Yellow Heart" },
      { src: heartGreen, alt: "Green Heart" },
    ];
    const container = createElement("div", {
        className: "container",
        children:
        [
            createElement("div",{
                className: "hearts",
                children: [
                  ...hearts.map((heart) => (
                    createElement("img", {
                          src: heart.src,
                          alt: heart.alt,
                          onclick: heartCounter
                      })
                  )),
                  createElement("img", {
                      id: "trash",
                      src: trashCan,
                      alt: "Delete everything!",
                      onclick: cleanStorage
                  }),
                ],
```

Are you sure you need  `id: "trash"`?

Reply:
way better.

Reply:
Alternative:

```suggestion
export function createFavorites () {
    const heartImages = [
      { src: heartOrange, alt: "Orange Heart", onclick: heartCounter },
      { src: heartBlue, alt: "Blue Heart", onclick: heartCounter },
      { src: heartPink, alt: "Pink Heart", onclick: heartCounter },
      { src: heartYellow, alt: "Yellow Heart", onclick: heartCounter },
      { src: heartGreen, alt: "Green Heart", onclick: heartCounter },
      {  id: "trash", src: trashCan, alt: "Delete everything!", onclick: cleanStorage }
    ];
    const container = createElement("div", {
        className: "container",
        children:
        [
            createElement("div",{
                className: "hearts",
                children: heartImages.map((heartImage) => (
                    createElement("img", heartImage)
                  ))
                ,
```

Reply:
in this specific case not, because i add the onclick directly to the object and not seperatly, the way to solve it with a map function is a good idea, way better to administrate thanks, will build it into the haiku.js also....